app/src/main/java/com/example/bigowlapp/activity/HomePageActivity.java

```diff
@@ -46,8 +54,55 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         initialize();
+        if (checkSelfPermission(Manifest.permission.READ_PHONE_STATE) != PackageManager.PERMISSION_GRANTED) {
+            requestPermissions(new String[]{Manifest.permission.READ_PHONE_STATE}, 10);
```

Comment:
I got issues with missing permissions, but I was able to fix it by modifying this line to this:

```java
requestPermissions(new String[]{Manifest.permission.READ_PHONE_STATE, Manifest.permission.READ_SMS, Manifest.permission.READ_PHONE_NUMBERS}, 10);
```

Reply:
I also added those permissions